### PR TITLE
WS-26: Login Internal API

### PR DESCRIPTION
**Pull request by michael-patsko**
Create authenticateLogin method in internalApiClient

### CODE REVIEW

File: frontend/src/components/login/LoginManager.tsx [hunk 1]
```diff
@@ -29,10 +30,16 @@ export const LoginManager: React.FunctionComponent = ({ children }) => {
...
+    authenticateLogin(username, password).then((didLogin) => {
+      if (didLogin) {
+        setUsername(username);
+        setPassword(password);
+        setLoggedIn(true);
+        setAdmin(true);
+      } else {
+        console.log("User could not be authenticated.");
+      }
+    });
```
Comment (TimLeach635) [inline frontend/src/components/login/LoginManager.tsx:42]: Oops! Sorry, I should have caught this before, but there should only be changes to `internalApiClient.ts` in this ticket. Could you remove those changes from here, and bring them forward into the new branch you're working on? That's potentially git-tricky so please get me to pair on it if you're stuck for even a second

File: frontend/src/clients/internalApiClient.ts [hunk 2]
```diff
@@ -35,3 +35,24 @@ export async function fetchSightings(): Promise<SightingListResponse> {
...
+  username: string,
+  password: string
+): Promise<boolean> {
+  const details = `${username}:${password}`;
+  const encodedDetails = btoa(details);
+  const authHeader = `Basic ${encodedDetails}`;
+  const response = await fetch(`https://localhost:5001/login`, {
+    method: "GET",
+    headers: {
+      Authorization: authHeader,
```
Comment (TimLeach635) [inline frontend/src/clients/internalApiClient.ts:49]: This is very nitpicky, but could we wrap this in speech marks?
```suggestion
      "Authorization": authHeader,
```
The reasoning I can't fully rationalise, but the field name in this case corresponds to a specifically formatted, "human"-readable string in the request (plus I've seen it done that way before).

Basically, the key name here isn't "just code", it's relevant in another context, so the specific formatting and spelling matters ergo string

**Review by TimLeach635 — CHANGES_REQUESTED** [review]
A little annoying change, sorry about that

File: frontend/src/clients/internalApiClient.ts (hunk 2 above)
Comment (michael-patsko) [reply frontend/src/clients/internalApiClient.ts:49]: Wrapped Authorization in quotation marks as requested

**Review by TimLeach635 — APPROVED** [review]

**Pull request merged**